File: app-android/app/src/main/java/com/health2u/data/remote/dto/AppointmentDto.kt
package com.health2u.data.remote.dto

import com.google.gson.annotations.SerializedName

data class AppointmentDto(
    @SerializedName("id") val id: String,
    @SerializedName("user_id") val userId: String,
    @SerializedName("title") val title: String,
    @SerializedName("description") val description: String?,
    @SerializedName("doctor_name") val doctorName: String?,
    @SerializedName("location") val location: String?,
    @SerializedName("date_time") val dateTime: Long,
    @SerializedName("reminder_minutes") val reminderMinutes: Int?,
    @SerializedName("status") val status: String,
    @SerializedName("created_at") val createdAt: Long
)

File: app-ios/Sources/Health2u/Data/Remote/DTOs/AppointmentDTO.swift
import Foundation

public struct AppointmentDTO: Codable, Equatable, Sendable {
    public let id: String
    public let userId: String
    public let title: String
    public let description: String?
    public let doctorName: String?
    public let location: String?
    public let dateTime: Date
    public let reminderMinutes: Int?
    public let status: String
    public let createdAt: Date

    public init(
        id: String,
        userId: String,
        title: String,
        description: String? = nil,
        doctorName: String? = nil,
        location: String? = nil,
        dateTime: Date,
        reminderMinutes: Int? = nil,
        status: String,
        createdAt: Date
    ) {
        self.id = id
        self.userId = userId
        self.title = title
        self.description = description
        self.doctorName = doctorName
        self.location = location
        self.dateTime = dateTime
        self.reminderMinutes = reminderMinutes
        self.status = status
        self.createdAt = createdAt
    }
}
